application/client.plugins/projects/serial/src/lib/views/dialog/components.ts

```diff
@@ -0,0 +1,87 @@
+// tslint:disable:no-inferrable-types
+
+import { Component, ChangeDetectorRef, Input, OnInit, OnDestroy, ViewChild } from '@angular/core';
+import { IPortInfo, IPortState } from '../../common/interface.portinfo';
+import { IOptions } from '../../common/interface.options';
+import { SidebarVerticalPortOptionsWriteComponent } from '../sidebar.vertical/port.options.write/component';
+
+interface IConnected {
+    port: IPortInfo;
+    options: IOptions;
+    state: IPortState;
+}
+
+@Component({
+    selector: 'lib-sb-port-dialog-com',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less']
+})
+
+export class SidebarVerticalPortDialogComponent implements OnInit, OnDestroy {
+
+    @ViewChild('optionsCom', {static: false}) _optionsCom: SidebarVerticalPortOptionsWriteComponent;
+
+    @Input() private _onConnect: () => void;
+    @Input() private _requestPortList: () => IPortInfo[];
+    @Input() private _getSelected: (IPortInfo) => void;
+    @Input() private _get_optionsCom: (SidebarVerticalPortOptionsWriteComponent) => void;
+    @Input() private _stopSpy: () => void;
+
+    @Input() public _ng_getState: (IPortInfo) => IPortState;
+    @Input() public _ng_canBeConnected: () => boolean;
+    @Input() public _ng_connected: IConnected[];
+    @Input() public _ng_isPortSelected: (port: IPortInfo) => boolean;
+    @Input() public _ng_onOptions: () => void;
+    @Input() public _ng_onPortSelect: (port: IPortInfo) => void;
```

Comment:
I put `_ng_` in front of those methods because I use them in the _template.html_ of the dialog window. 
Because those functions modify variables located in sidebar.vertical/components.ts, I use them as callback function to get new status on user interaction and return it.
About the service, I will start to think about a solution